src/redux/thunks/weather.ts

```diff
@@ -1,54 +1,67 @@
 import { isAxiosError } from 'axios';
+import i18next from '../../services/localization/i18n';
 import type { ThunkType } from '../../utils/types/common';
 import type { WeatherAction } from '../types/weather';
 import type { IWeather } from '../../utils/types/api/weather';
 import { getDegeocodingCityAPI, getWeatherAPI } from '../../services/api/weather/weather';
 import { setCity, setLocation, setWeatherDataSuccess, setWeatherDataFailure } from '../actions/weather';
-import { StatusCode, LocalStorageKey } from '../../utils/types/enums';
+import { StatusCode, LocalStorageKey, ContentTxtKey, ErrorTxtKey } from '../../utils/types/enums';
 import { getLocalItem, removeLocalItem } from '../../services/browserDataStorage/localStorage';
 
 export const setCityWithLocation = (city: string, location: string): ThunkType<WeatherAction> => {
   return async (dispatch) => {
     const newCity = getLocalItem<string>(LocalStorageKey.City) || city;
-    dispatch(setCity(newCity));
+
+    if (newCity === 'Минск' || newCity === 'Minsk') {
+      dispatch(setCity(city));
+    } else {
+      dispatch(setCity(newCity));
+    }
+
     dispatch(setLocation(location));
```

Comment:
When we add new language - we need to update the condition as well, so it is hard to maintain
Let's try find an util which may return list of translations for the city or refer to a city id instead, or any other possible solution

Reply:
According to my logic:
1.	If user allows access to the location when loading the page and everything goes well, he gets the city name considering the language mode (considered in the API).
2.	If user for some reason does not receive the location, he receives "Minsk" city (at the first load).
3.	I assume when user enters the city, he does it in the language he needs. Then this value goes to the local storage. In the future after reloading the page or switching the language user gets the entered city name.

This condition is needed to switch the language at the first load if the location has not received and while user has not entered the city name. And as a nice addition, users who themselves entered "Minsk" city can get the city translation. I do not plan to update this condition.

According to my logic, I do not want to translate the city entered by user. This condition is not required, but rather a nice bonus. I could not find exactly those utilities which you wrote to me about. If there is something similar, then it complicates the translation logic, as well as the logic in general.

My logic is undeniable :)